ocrd_tesserocr/recognize.py

```diff
@@ -56,26 +67,65 @@ def __init__(self, *args, **kwargs):
-        (remove any existing segmentation below the line level, and)
-        create new segmentation below the line level if necessary.
+        ``textequiv_level`` if it exists, and if each level's respective
+        ``overwrite_LEVEL`` is disabled. 
+        Otherwise stop at the highest level with ``overwrite_LEVEL==True``,
+        and remove any existing segmentation at that level.
         
         Set up Tesseract to recognise each segment's image (either from
         AlternativeImage or cropping the bounding box rectangle and masking
         it from the polygon outline) with the appropriate mode and ``model``.
         
+        Next, iterate over the result hierarchy from the current level
+        in the PAGE hierarchy down to the requested ``textequiv_level``,
+        creating new segmentation at each level.
+        
         Put text and confidence results into the TextEquiv at ``textequiv_level``,
         removing any existing TextEquiv.
         
-        Finally, make the higher levels consistent with these results by concatenation,
-        ordered as appropriate for its readingDirection, textLineOrder, and ReadingOrder,
-        and joined by whitespace, as appropriate for the respective level and Relation/join
-        status.
+        Finally, make all higher levels consistent with these text results
+        by concatenation, ordering according to each level's respective
+        readingDirection, textLineOrder, and ReadingOrder, and joining
+        by whitespace as appropriate for each level and according to its
+        Relation/join status.
+        
+        The special value ``none`` for ``textequiv_level`` behaves like ``word``,
+        except that no actual text recognition will be performed, only layout
+        analysis.
+        
+        In other words:
+        - If ``overwrite_regions``, segment regions, else iterate existing regions.
+        - If ``textequiv_level==region``, then recognize text in the region,
+          annotate it, and continue with the next region. Otherwise...
+        - If ``overwrite_lines``, segment lines, else iterate existing text lines.
+        - If ``textequiv_level==line``, then recognize text in the text lines,
+          annotate it, and continue with the next line. Otherwise...
+        - If ``overwrite_words``, segment words, else iterate existing words.
+        - If ``textequiv_level==word``, then recognize text in the words,
+          annotate it, and continue with the next word. Otherwise...
+        - If ``textequiv_level==glyph``, then recognize text in the glyphs and
+          continue with the next glyph. Otherwise...
+        - (i.e. ``none``) annotate no text and be done with segmentation.
+        
+        Thus, enabling the ``overwrite_*`` modes makes this processor behave more
+        like a segmentation processor, and setting ``textequiv_level`` to ``none``
+        makes it a segmentation-only processor.
```

Comment:
OK, just so I understand this, `recognize -P overwrite_lines -P textequiv_level none` is equivalent to `segment-lines` now, correct?

Reply:
yes, see comment on PR :tada:

Reply:
Yes, exactly!

And if you add `-P overwrite_regions true` you get the `ocrd-cis-ocropy-segment` and `sbb-textline-detector` behaviour.